compiler/rustc_passes/src/stability.rs

```diff
@@ -559,6 +559,24 @@ impl<'tcx> MissingStabilityAnnotations<'tcx> {
             }
         }
     }
+
+    fn check_private_stability(&self, hir_id: HirId, span: Span) {
+        let stab = self.tcx.stability().local_stability(hir_id);
```

Comment:
I don't think this is the right check; try using [`lookup_stability`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyCtxt.html#method.lookup_stability) instead.
```suggestion
        let stab = self.tcx.lookup_stability(hir_id);
```